src/test/java/org/broadinstitute/consent/http/authentication/UserAuthorizerTest.java

```diff
@@ -43,8 +43,14 @@ void testAuthorizeNotAuthorized() {
   @Test
   void testAuthorizeAuthorized() {
     when(userRoleDAO.findRoleNamesByUserEmail(any()))
-      .thenReturn(List.of(UserRoles.CHAIRPERSON.getRoleName()));
+        .thenReturn(List.of(UserRoles.CHAIRPERSON.getRoleName()));
     assertTrue(authorizer.authorize(authorizedUser, Resource.CHAIRPERSON, context));
   }
 
+  @Test
+  void testAuthorizedServiceAccount() {
+    when(userRoleDAO.findRoleNamesByUserEmail(any())).thenReturn(
+        List.of(UserRoles.SERVICE_ACCOUNT.getRoleName()));
+    assertTrue(authorizer.authorize(authorizedUser, Resource.SERVICE_ACCOUNT, context));
```

Comment:
Instead of mocking `AuthUser`, why not create one? I think it's better to not mock POJOs if we can avoid it.
```suggestion
    AuthUser user = new AuthUser("email");
    when(userRoleDAO.findRoleNamesByUserEmail(user.getEmail()))
        .thenReturn(List.of(UserRoles.SERVICE_ACCOUNT.getRoleName()));
    assertTrue(authorizer.authorize(user, Resource.SERVICE_ACCOUNT, context));
```

Reply:
I reduced your suggestion SLOC by 33%. :-). No need to create the AuthUser at all.

Reply:
The problem is that this doesn't validate the argument to `findRoleNamesByUserEmail()`. If the method being tested passes a different string to that method, this test doesn't detect that error. Creating an object with a value to verify allows us to make sure that this part of the code is working as expected.

Although I don't see what coverage this provides over `testAuthorizeAuthorized()`. If you want to test all roles, you could make this a parameterized test that tests all `UserRoles` members. But since there's no difference in how these values are handled, I don't think there's a benefit for that additional testing.